src/ICSharpCode.SharpZipLib/Zip/Compression/Streams/InflaterInputStream.cs

```diff
@@ -301,6 +323,19 @@ public ICryptoTransform CryptoTransform
 			}
 		}
 
+		private int CalculateDecryptionSize(int availableBufferSize)
+		{
+			int size = DecryptionLimit ?? availableBufferSize;
+			size = Math.Min(size, availableBufferSize);
+			
+			if (DecryptionLimit.HasValue)
+			{
+				DecryptionLimit -= size;
+			}
+
+			return size;
```

Comment:
```suggestion
			if (!DecryptionLimit.HasValue)
			{
			    return availableBufferSize;
			}

			var size = Math.Min(DecryptionLimit, availableBufferSize);
			DecryptionLimit -= size;

			return size;
```

Does look simpler to me, as three of the operations only apply when it's not null.

Reply:
Changed